cli/flox/src/utils/metrics.rs

```diff
@@ -246,36 +248,55 @@ impl MetricsBuffer {
             .unwrap_or(false)
     }
 
+    /// Overwrites the buffer file with the current in-memory buffer contents
+    fn overwrite_file(&mut self) -> Result<()> {
+        self.storage
+            .set_len(0)
+            .context("Could not truncate metrics buffer file")?;
+        for entry in &self.buffer {
+            let mut buffer_json = String::new();
+            buffer_json.push_str(&serde_json::to_string(entry)?);
+            buffer_json.push('\n');
+            self.storage
+                .write_all(buffer_json.as_bytes())
+                .context("could not write metrics entry to buffer file")?;
+        }
+        self.storage.flush()?;
+        Ok(())
+    }
+
     /// Pushes a new metric entry to the buffer and syncs it to the buffer file
     fn push(&mut self, entry: MetricEntry) -> Result<()> {
         debug!("pushing entry to metrics buffer: {entry:?}");
 
-        // update file with new entry
-        // [MetricsBuffer::read] ensures that the file is opened with write permissions
-        // and append mode.
-        let mut buffer_json = String::new();
-        buffer_json.push_str(&serde_json::to_string(&entry)?);
-        buffer_json.push('\n');
-        self.storage
-            .write_all(buffer_json.as_bytes())
-            .context("could not write new metrics entry to buffer file")?;
-        self.storage.flush()?;
+        // Enforce maximum buffer size by removing oldest entry if at capacity
+        let needs_rewrite = if self.buffer.len() >= MAX_BUFFER_SIZE {
```

Comment:
question: do we care about files that are _already_ larger than `MAX_BUFFER_SIZE + 1`?.
Right now this only seems to maintain size not truncate down to `MAX_BUFFER_SIZE`. That will work fine for new files that only add one at a time but existing larger files will remain large (though giving us a chance to get the data via batched transfer...)